intellij/src/saros/intellij/ui/actions/ConnectServerAction.java

```diff
@@ -51,37 +59,123 @@ public void execute() {
+
+    } catch (IllegalAWTContextException e) {
+      LOG.error("Failed to show question on whether to really switch accounts to user", e);
+
+      NotificationPanel.showError(
+          MessageFormat.format(
+              Messages.ConnectServerAction_leave_session_confirmation_host_addendum_error_message,
+              e.getMessage()),
+          Messages.ConnectServerAction_leave_session_confirmation_host_addendum_error_title);
+
+      return;
+    }
+
+    if (reallyReconnect) {
+      connectToAccount(account, true);
+    }
   }
 
   /**
-   * Connects an Account to the XMPPService and sets it as active.
+   * Connects to the given XMPP account.
+   *
+   * <p>If the information is passed that there already is an existing connection, the previous
+   * connection will be ended before connecting to the new account. This will also end any existing
+   * sessions if present.
    *
    * @param account the account to connect to
+   * @param isAlreadyConnected whether there currently is an existing connection
    */
-  private void connectAccount(final XMPPAccount account) {
-    ProgressManager.getInstance()
-        .run(
-            new Task.Modal(project, "Connecting...", false) {
-
-              @Override
-              public void run(@NotNull ProgressIndicator indicator) {
-
-                LOG.info(
-                    "Connecting server: ["
-                        + account.getUsername()
-                        + "@"
-                        + account.getServer()
-                        + "]");
-
-                indicator.setIndeterminate(true);
-
-                try {
-                  connectionHandler.connect(account, false);
-                } finally {
-                  indicator.stop();
-                }
-              }
-            });
+  private void connectToAccount(@NotNull XMPPAccount account, boolean isAlreadyConnected) {
+
+    ThreadUtils.runSafeAsync(
+        "Connecting to XMPP account " + account,
+        LOG,
+        () -> {
+          if (isAlreadyConnected) {
+            connectionHandler.disconnect();
+          }
+
+          String quailifiedAccountName = account.getUsername() + '@' + account.getDomain();
+
+          String message =
+              MessageFormat.format(
+                  Messages.ConnectServerAction_progress_message, quailifiedAccountName);
+
+          ProgressManager.getInstance()
+              .run(
+                  new Task.Modal(project, message, false) {
+
+                    @Override
+                    public void run(@NotNull ProgressIndicator indicator) {
+
+                      indicator.setIndeterminate(true);
+
+                      LOG.info("Connecting to account: [" + quailifiedAccountName + "]");
+
+                      try {
+                        connectionHandler.connect(account, false);
+                      } finally {
+                        indicator.stop();
+                      }
+                    }
```

Comment:
catch(Exception e) {log(e)} otherwise I do not know if it is logged at all, at least it could not be present in the Saros Log file.

Reply:
The IntelliJ API has a good track-record on re-throwing exceptions outside of the runnable, in which case they would be caught by the `ThreadUtils.runSafeAsync(...)`.

Catching them ourselves is fine as well, but a blanket catch statement  (`catch(Exception e) {`) seems overkill to me. Are there any exceptions we are worried about in particular? The javadoc doesn't specify anything and I couldn't find anything from having a quick scroll through the code.